784xx Release Notes - CI (#677)
harness/developer-hub: release-notes/continuous-integration.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## February 14, 2023, version 784xx <!--add final version number & date -->

### What's new

* Microsoft Windows (amd64) is now a supported [Harness Cloud](../docs/continuous-integration/ci-quickstarts/hosted-builds-on-virtual-machines-quickstart.md) build infrastructure option. (CI-5455)

> * Added **Python** as a built-in **Shell** option for [Run steps](../docs/continuous-integration/ci-technical-reference/run-step-settings.md). (CI-6692)

Reviewer: ```suggestion
* **Python** is now available as a built-in **Shell** option for [Run steps](../docs/continuous-integration/ci-technical-reference/run-step-settings.md). (CI-6692)
```

* [Run steps](../docs/continuous-integration/ci-technical-reference/run-step-settings.md) can now reference images in GCR through [GCP connectors](../docs/platform/7_Connectors/connect-to-google-cloud-platform-gcp.md) that use inherited credentials.

>   * GCP connectors are authenticated through either a GCP service account key or by inheriting credentials from the Harness delegate running in GCP. This change improves how you can use GCP connectors with inherited credentials in your pipelines.

Reviewer: ```suggestion
  * GCP connectors are now authenticated through either a GCP service account key or by inheriting credentials from the Harness delegate running in GCP. This change improves how you can use GCP connectors with inherited credentials in your pipelines.
```

Author: They already were authenticated by both methods, but Run steps couldn't reference images in GCR if the GCP connector had inherited credentials. It would fail with an error that "Inherit FromDelegate" was not supported.

>   * Previously, if you wanted a Run step to call an image in GRC, the GCP connector attached to your Run step had to use service account key authentication. Now, the GCP connector can use either authentication method. (CI-5758, ZD-38986)

Author: They are sub-bullets under line 22

>   * Previously, if you wanted a Run step to call an image in GRC, the GCP connector attached to your Run step had to use service account key authentication. Now, the GCP connector can use either authentication method. (CI-5758, ZD-38986)

Author: I moved the ids to be on the first level bullet.

> * Use [Background steps](../docs/continuous-integration/ci-technical-reference/background-step-settings.md) to configure service dependencies.

Reviewer: ```suggestion
* [Background steps](../docs/continuous-integration/ci-technical-reference/background-step-settings.md) can now be used to configure service dependencies.
```

Author: Technically this was introduced in November. This is kind of a re-announcement... The actual change this time is that they've removed Configure Service Dependency from the visual editor for new pipelines.

  * The Background step allows for better control and configuration of services than the now-deprecated Configure Service Dependency step.
  * Pipelines with Configure Service Dependency steps remain backwards compatible, but this step is not available for new pipelines.
  * Replace Configure Service Dependency steps with Background steps to take advantage of the more robust control and configuration option. (CI-5580)
* Pipeline execution status links in Git pull requests now direct you to the associated stage within the pipeline, rather than the pipeline as a whole. (CI-6813)

> * Improved handling of Azure repo URLs in [Git webhook pipeline triggers](../docs/platform/11_Triggers/triggering-pipelines.md). (CI-5720)

Reviewer: Can you provide some details on what was improved? And is this a fix or a new item? Seems like it might be a fix.

Author: It's an enhancement. I have asked multiple times and I haven't gotten any specifics. I think this is because the changes are about how the API handles these URLs on the backend than something that would usually be exposed to the user.

There is an old and new URL format for Azure repos. The ticket was originally created due to an auth issue with webhook triggers that received old format URLs. Instead of fix that one issue, my understanding is that multiple changes were made to improve handling of these old format URLs in webhook triggers.

### Early access

This release does not include early access features.

### Deprecations

The Configure Service Dependency step is deprecated in favor of the [Background step](../docs/continuous-integration/ci-technical-reference/background-step-settings.md), which allows for better control and configuration of services. Pipelines with Configure Service Dependency steps remain backwards compatible, but this step is not available for new pipelines. Replace Configure Service Dependency steps with Background steps to take advantage of the more robust control and configuration options. (CI-5580)

### Fixed issues

> * With the feature flag `NEW_LEFT_NAV_BAR` enabled, you can view build details at the project level without drilling down into individual modules. When viewing build details from project level, the **Commits** tab produced a `404` error. This is fixed. (CI-6817)

Reviewer: ```suggestion
* With the feature flag `NEW_LEFT_NAV_BAR` enabled, you can view build details at the project level without drilling down into individual modules. When viewing build details from the project level, the **Commits** tab produced a `404` error. This is fixed. (CI-6817)
```

* Expressions used in pipeline codebase properties unexpectedly resolved to `null`, which caused builds to fail due to missing variable input. These expressions now resolve to their expected values. (CI-6679, ZD-38122, ZD-38241, ZD-38698, ZD-39088)
   ![The CI Codebase Configuration window with the fields set to accept variable expression input.](static/ci-codebase-config-exp-values.png)

> * The CI Getting Started workflow leads you through creating an SCM connector and a pipeline. We fixed an issue where exiting the Getting Started workflow before creating a pipeline resulted in incomplete connector configuration. Attempting to use the incomplete connector in a pipeline produced the following error: `Invalid argument(s): Both plain text and secret value cannot be null for the field`. If you encounter this error, replace the pipeline's SCM connector with a new one. (CI-6443)

Reviewer: Avoid "We"

* Added **Python** as a built-in **Shell** option for [Run steps](../docs/continuous-integration/ci-technical-reference/run-step-settings.md). (CI-6692)
* [GCP connectors](../docs/platform/7_Connectors/connect-to-google-cloud-platform-gcp.md) are authenticated through either a GCP service account key or by inheriting credentials from the Harness delegate running in GCP. Run steps can now reference images in GCR through GCP connectors with inherited credentials. (CI-5758, ZD-38986)
* Improved handling of Azure repo URLs in Git webhook pipeline triggers. (CI-5720) <!--not clear-->
* Microsoft Windows (amd64) is now a supported Harness Cloud build infrastructure option. (CI-5455)
* Use [Background steps](../docs/continuous-integration/ci-technical-reference/background-step-settings.md) to configure service dependencies.
  * The Background step allows for better control and configuration of services than the now-deprecated Configure Service Dependency step.
  * Pipelines with Configure Service Dependency steps remain backwards compatible, but this step is not available for new pipelines.

>   * Replace Configure Service Dependency steps with Background steps to take advantage of the more robust control and configuration option. (CI-5580)

Reviewer: @aimurphy I know this is your first release notes PR at Harness, so please review [ ](https://harness.atlassian.net/wiki/spaces/DOCS/pages/1689845830/Release+notes+process+and+guidelines#What%E2%80%99s-new) and update these entries for new features to make them more descriptive and include the benefits. You may need to work with Pranav on some of these. Make sure there's a link to associated docs. 

Also, every entry in a module's "What's new" section must also be added to the overall What's new topic [ ](https://developer.harness.io/release-notes/whats-new). Michael made some change to the overall What's new in [ ](https://github.com/harness/developer-hub/pull/678/files) so make sure to coordinate on updating that file to avoid merge conflicts.

Reviewer: https://harness.atlassian.net/wiki/spaces/DOCS/pages/1689845830/Release+notes+process+and+guidelines

I just realized the above link was not included in my original comment.

Author: Discussed w/ PM & got approval

### Early access

> No early access features added in this release.

Author: fixed

### Deprecations

The Configure Service Dependency step is deprecated in favor of the [Background step](../docs/continuous-integration/ci-technical-reference/background-step-settings.md), which allows for better control and configuration of services. Pipelines with Configure Service Dependency steps remain backwards compatible, but this step is not available for new pipelines. Replace Configure Service Dependency steps with Background steps to take advantage of the more robust control and configuration options. (CI-5580)

### Fixed issues

> * With the NEW_LEFT_NAV_BAR feature flag enabled, you can view builds details at the project level without drilling down into individual modules. We fixed an issue where the **Commits** tab produced a `404` error when viewing build details from the project level. (CI-6817)

Reviewer: ```suggestion
* With the NEW_LEFT_NAV_BAR feature flag enabled, you can view build details at the project level without drilling down into individual modules. We fixed an issue where the **Commits** tab produced a `404` error when viewing build details from the project level. (CI-6817)
```

> * With the NEW_LEFT_NAV_BAR feature flag enabled, you can view builds details at the project level without drilling down into individual modules. We fixed an issue where the **Commits** tab produced a `404` error when viewing build details from the project level. (CI-6817)

Reviewer: Are these two different issues in the same issue? Not sure why the first sentence is part of the fixed issue description. Also, avoid "we" per MSTP [ ](https://learn.microsoft.com/en-us/style-guide/grammar/person#avoid-plural-first-person-we-us)

Author: The issue was only happening with that FF enabled, because you had to go through the "Projects" tab that is part of that FF.

> * We fixed an issue where expressions used in a pipeline's codebase properties resolved to `null` rather than their expected values. (CI-6679, ZD-38122, ZD-38241, ZD-38698, ZD-39088)

Reviewer: Same comment as above about "we".

### Early access features

> No early access features added in this release.

Reviewer: The original is correct.

Author: fixed it back.

### What's new

* This release includes two new CI steps for integrating your Harness CI pipelines with GitHub Actions and Bitrise. The following steps are available in Harness Cloud build infrastructures only.
   - An Actions step to run GitHub Actions.
   - A Bitrise step to run Bitrise steps and workflows. (CI-6479)

> * Harness CI now supports remote debugging of remote builds in Harness Cloud, Kubernetes, and VMs in AWS, Azure, and other cloud platforms. If a build fails at a Run step, you can rerun the build in debug mode. This option is available in the **Builds**, **Execution**, and **Execution History** pages of the Harness UI. (CI-6350)

Reviewer 2: @aimurphy you remove Remote Debugging from this list.

Author: Remove the whole bullet?

### Early access features

> No early access features added in this release.

Reviewer: The original is correct. Undo all changes to this sentence in this file.
